### PR TITLE
Bump black version and relax constraint

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary
This commit bumps the black version we pin to the latest release,
22.1.0. This release is also the first release not marked as beta and
with that black has introduced a stability policy where no formatting
changes will be introduced on a major version release (which is the
year). [1] With this new policy in place we no longer need to pin to a
single version and can instead constrain the requirement to just the
major version without worrying about a new release breaking ci or local
development. This commit does that and sets the black version
requirement to be any 22.x.y release so that we'll continue to get
bugfixes moving forward without having to manually bump a pinned version.

### Details and comments

[1] https://black.readthedocs.io/en/latest/the_black_code_style/index.html#stability-policy

### CODE REVIEW

File: qiskit/compiler/transpiler.py [hunk 1]
```diff
@@ -331,16 +332,43 @@ def callback_func(**kwargs):
...
-    # Transpile circuits in parallel
-    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_args)))
+    # Avoid parallel processing overhead if only one circuit
+    if len(circuits) == 1:
+        out_circuits = [_transpile_circuit(circuits[0], transpile_args[0])]
+    else:
+        # Runtime import to avoid cycles and potential overhead on initial import
+        from qiskit.circuit.qpy_serialization import dump, load
+
+        # Transpile circuits in parallel
```
Comment (jakelishman) [inline qiskit/compiler/transpiler.py:342]: What's going on here?

Comment (mtreinish) [reply qiskit/compiler/transpiler.py:342]: ugh, I had a dirty checkout (where I was testing using qpy instead of pickle to serialize circuits for multiprocessing) when I wrote the PR and didn't realize it before committing. One sec I'll fix this now

Comment (mtreinish) [reply qiskit/compiler/transpiler.py]: I just forced push a new branch that removed this change

File: qiskit/tools/parallel.py [hunk 2]
```diff
@@ -65,9 +65,6 @@
     # Default False on Windows
     if sys.platform == "win32":
         PARALLEL_DEFAULT = False
-    # On python 3.9 default false to avoid deadlock issues
-    elif sys.version_info[0] == 3 and sys.version_info[1] == 9:
-        PARALLEL_DEFAULT = False
```
Comment (jakelishman) [inline qiskit/tools/parallel.py:70]: I think this is unrelated too (and at a glance it should presumably have been `sys.version_info[1] >= 9` anyway haha).

**Review by jakelishman — COMMENTED** [review]
this time I spotted it by wondering why the diff was asymmetrical - +445/-448 or something.  But that's all I could see.

File: qiskit/tools/parallel.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/tools/parallel.py:70]: Sigh, yeah that was also in my local experiment that I was playing with (the issue with python 3.9 is pickle would just occasionally hang when serializing a parameterized circuit, which was an issue for parallel transpile so I was trying to leverage qpy to workaround that). I'll fix this now too. I feel like I should just make a fresh checkout and start from scratch on this branch :)

Comment (mtreinish) [reply qiskit/tools/parallel.py]: Ok, this time I hope I got it right, started from a fresh checkout and reran it

**Review by jakelishman — APPROVED** [review]
